File: src/main/java/kr/co/lunatalk/domain/category/repository/CategoryRepository.java
package kr.co.lunatalk.domain.category.repository;

import kr.co.lunatalk.domain.category.domain.Category;
import kr.co.lunatalk.domain.category.domain.CategoryStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface CategoryRepository extends JpaRepository<Category, Long>,CategoryRepositoryCustom {

	Optional<Category> findByName(String name);
	boolean existsByName(String name);

	List<Category> findAllByStatus(CategoryStatus status);
}

File: src/main/kotlin/kr/co/lunatalk/domain/category/repository/CategoryRepository.kt
package kr.co.lunatalk.domain.category.repository

import kr.co.lunatalk.domain.category.domain.Category
import kr.co.lunatalk.domain.category.domain.CategoryStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface CategoryRepository : JpaRepository<Category, Long>, CategoryRepositoryCustom {
    fun findByName(name: String): Optional<Category>
    fun existsByName(name: String): Boolean
    fun findAllByStatus(status: CategoryStatus): List<Category>
    fun findByIdAndStatus(id: Long, status: CategoryStatus): Optional<Category>
}
